src/Admin/AbstractAdmin.php

```diff
@@ -1008,37 +920,19 @@ public function getActiveSubClass()
     public function getActiveSubclassCode()
     {
         if (!$this->hasActiveSubClass()) {
-            @trigger_error(sprintf(
-                'Calling %s() when there is no active subclass is deprecated since sonata-project/admin-bundle 3.52 and will throw an exception in 4.0. '.
-                'Use %s::hasActiveSubClass() to know if there is an active subclass.',
-                __METHOD__,
-                __CLASS__
-            ), E_USER_DEPRECATED);
-            // NEXT_MAJOR : remove the previous `trigger_error()` call, the `return null` statement, uncomment the following exception and declare string as return type
-            // throw new \LogicException(sprintf(
-            //    'Admin "%s" has no active subclass.',
-            //    static::class
-            // ));
-
-            return null;
+            throw new \LogicException(sprintf(
```

Comment:
We should go here with our own LogicException and I really would love to see the way doctrine uses exceptions: named ctors, like:

LogicException::noActiveSubClass($class)

WDYT ?